internal/utils/msa/msa.go

```diff
@@ -0,0 +1,120 @@
+// Copyright The Enterprise Contract Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+// Not sure if this is idiomatic go, but I'm trying to make it easier
+// to work with untyped maps such as might be created by unmarshalling
+// arbitrary JSON data
+
+// Short for MapStringAny
+package msa
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type MapStringAny map[string]any
+
+// Create a MapStringAny from a plain map[string]any
+func FromMap(m map[string]any) MapStringAny {
+	return MapStringAny(m)
+}
+
+// Create a MapStringAny by unmarshaling json bytes
+func FromJsonBytes(bytes []byte) (MapStringAny, error) {
+	var m MapStringAny
+	err := json.Unmarshal(bytes, &m)
+	return m, err
+}
+
+// Create a MapStringAny by unmarshaling a json string
+func FromJson(str string) (MapStringAny, error) {
+	return FromJsonBytes([]byte(str))
+}
+
+// Convert back to a plain map[string]any
+func (m MapStringAny) ToMap() map[string]any {
+	return map[string]any(m)
+}
+
+// Marshal to json bytes with optional indenting
+func (m MapStringAny) ToJSON(args ...string) []byte {
+	var (
+		result []byte
+		err    error
+	)
+
+	if len(args) > 0 {
+		result, err = json.MarshalIndent(m, "", args[0])
+	} else {
+		result, err = json.Marshal(m)
+	}
+	if err != nil {
+		panic(fmt.Errorf("Unexpected marshalling problem: %w", err))
+	}
+	return result
+}
+
+// Marshal to a json string with optional indenting
+func (m MapStringAny) ToJSONStr(args ...string) string {
+	return string(m.ToJSON(args...))
+}
+
+// Get a string value from a map with string keys
+func (m MapStringAny) GetStr(key string) (string, error) {
+	return mapGet[string](m, key)
+}
+
+// Get a bool value from a map with string keys
+func (m MapStringAny) GetBool(key string) (bool, error) {
+	return mapGet[bool](m, key)
+}
+
+// Get a float64 value from a map with string keys
+// (You might expect it to be an int, but a float64 is what
+// you get from the json decoder.)
+func (m MapStringAny) GetNum(key string) (float64, error) {
+	return mapGet[float64](m, key)
+}
+
+// Get a list from a map with string keys
+func (m MapStringAny) GetList(key string) ([]any, error) {
+	return mapGet[[]any](m, key)
+}
+
+// Get a map from a map with string keys
+func (m MapStringAny) GetMap(key string) (map[string]any, error) {
```

Comment:
This might have better utility

```suggestion
func (m JSONObject) Property(key string) (JSONObject, error) {
```

Reply:
I'll make a separate method for that.

Reply:
Done.